<?xml version="1.0" ?>

<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">

    <services>
       <service id="security.authentication.listener.symfony_connect" class="SymfonyCorp\Connect\Security\Firewall\ConnectAuthenticationListener" parent="security.authentication.listener.abstract" abstract="true">
            <call method="setOAuthConsumer">
                <argument type="service" id="symfony_connect.oauth_consumer" />
            </call>
            <call method="setOAuthCallback">
                <argument>symfony_connect_callback</argument>
            </call>
            <call method="setApi">
                <argument type="service" id="symfony_connect.api" />
            </call>
            <call method="setRethrowException">
                <argument>%kernel.debug%</argument>
            </call>
       </service>

       <service id="security.authentication.provider.symfony_connect" class="SymfonyCorp\Connect\Security\Authentication\Provider\ConnectAuthenticationProvider" abstract="true">
           <argument /> <!-- user provider -->
           <argument /> <!-- providerKey -->
       </service>

        <service id="security.authentication.entry_point.symfony_connect" class="SymfonyCorp\Connect\Security\EntryPoint\ConnectEntryPoint">
            <argument type="service" id="symfony_connect.oauth_consumer" />
            <argument type="service" id="security.http_utils" />
            <argument>symfony_connect_callback</argument>
        </service>

        <service id="security.user.provider.symfony_connect_in_memory" class="SymfonyCorp\Connect\Security\User\ConnectInMemoryUserProvider" abstract="true">
        </service>

        <service id="symfony_connect.oauth_controller" class="SymfonyCorp\Bundle\ConnectBundle\Controller\OAuthController">
            <argument type="service" id="security.authentication.entry_point.symfony_connect" />
            <tag name="controller.service_arguments" />
        </service>

        <service id="SymfonyCorp\Bundle\ConnectBundle\Controller\OAuthController" alias="symfony_connect.oauth_controller" public="true" />

        <service id="symfony_connect.authenticator" class="SymfonyCorp\Connect\Security\ConnectAuthenticator">
            <argument type="service" id="symfony_connect.oauth_consumer" />
            <argument type="service" id="symfony_connect.api" />
            <argument /> <!-- user provider -->
            <argument type="service" id="security.http_utils" />
            <argument type="service" id="logger" />
            <argument>symfony_connect_callback</argument>
            <argument type="service" id="twig" on-invalid="null" />
            <argument>%symfony_connect.api.start_template%</argument>
            <call method="setRethrowException">
                <argument>%kernel.debug%</argument>
            </call>
        </service>

    </services>
</container>
